test/resvg_test.exs

```diff
@@ -1,5 +1,11 @@
 defmodule Resvg.Test do
   use ExUnit.Case
+  # This external package is just a small single-module helper
```

Comment:
Could you expand on this? Are you talking about showing the generated HTML in the continuous integration somehow? What about if we had an "index HTML" near the root of the project that would link to all the failing diffs?

Reply:
That's more like a convenient thing you could add  to Approval lib.
But my idea is the following.
When the tests are running in the github CI if an approval is failing it produce nice html file when you can see the problem.
The idea is to add an extra step in the  github CI to check if some files where produced and if so keep them as artefact after that you can click on the CI ui to open them.
```
      - name: Check if HTML files exist
        id: check_html_files
        run: |
          if [ -n "$(find test-output/ -name '*.html' -type f 2>/dev/null)" ]; then
            echo "HTML files found"
            echo "::set-output name=html_exists::true"
          else
            echo "No HTML files found"
            echo "::set-output name=html_exists::false"
          fi

      - name: Upload HTML artifacts if they exist
        if: steps.check_html_files.outputs.html_exists == 'true'
        uses: actions/upload-artifact@v3
        with:
          name: test-failure-artifacts
          path: test-output/
```
[workflow github ci](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/storing-and-sharing-data-from-a-workflow)

Reply:
So basically creating an HTML artifact containing the diffs of all the images that are different so that they can be compared?

Reply:
It could be several HTML artifacts since they are already produced by approval as `reference_path <> "diff.html"`
It's just a suggestion don't spend too much time on this if you don't think it's worth.